evalml/model_understanding/permutation_importance.py

```diff
@@ -0,0 +1,181 @@
+    if not objective.is_defined_for_problem_type(pipeline.problem_type):
+        raise ValueError(f"Given objective '{objective.name}' cannot be used with '{pipeline.name}'")
+
+    if pipeline._supports_fast_permutation_importance:
+        perm_importance = _fast_permutation_importance(pipeline, X, y, objective,
+                                                       n_repeats=n_repeats,
+                                                       n_jobs=n_jobs,
+                                                       random_seed=random_seed)
+    else:
+        perm_importance = _slow_permutation_importance(pipeline, X, y, objective,
+                                                       n_repeats=n_repeats,
+                                                       n_jobs=n_jobs,
+                                                       random_seed=random_seed)
+
+    mean_perm_importance = perm_importance["importances_mean"]
+    feature_names = list(X.columns)
+    mean_perm_importance = list(zip(feature_names, mean_perm_importance))
+    mean_perm_importance.sort(key=lambda x: x[1], reverse=True)
+    return pd.DataFrame(mean_perm_importance, columns=["feature", "importance"])
+
+
+def calculate_permutation_importance_one_column(X, y, pipeline, col_name, objective,
+                                                n_repeats=5, fast=False, precomputed_features=None, random_seed=0):
+    """Calculates permutation importance for one column in the original dataframe.
+
+    Arguments:
+        pipeline (PipelineBase or subclass): Fitted pipeline
+        X (pd.DataFrame): The input data used to score and compute permutation importance
+        y (pd.Series): The target data
+        objective (str, ObjectiveBase): Objective to score on
+        n_repeats (int): Number of times to permute a feature. Defaults to 5.
+        fast (bool): Whether to use the fast method of calculating the permutation importance or not. Defaults to True.
+        precomputed_features (pd.DataFrame): Precomputed features necessary to calculate permutation importance using the fast method. Defaults to None.
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+
+    Returns:
+        float: Mean feature importance scores over a number of shuffles.
+    """
+    X = infer_feature_types(X)
+    y = infer_feature_types(y)
+    objective = get_objective(objective, return_instance=True)
+
+    if fast:
+        if precomputed_features is None:
+            raise ValueError("Fast method of calculating permutation importance requires precomputed_features")
+        if is_classification(pipeline.problem_type):
+            y = pipeline._encode_targets(y)
+        baseline_score = _fast_scorer(pipeline, precomputed_features, X, y, objective)
+        scores = _calculate_permutation_scores_fast(
+            pipeline, precomputed_features, y, objective, col_name, random_seed, n_repeats, _fast_scorer, baseline_score,
+        )
```

Comment:
@chukarsten 
Hmmm yeah I think there is a bunch of overlap, tldr being the one col version uses `_calculate_permutation_scores_fast` while `calculate_permutation_importance` uses `_fast_permutation_importance` (which calls in parallel `_fast_permutation_importance`)...
- one col expects the precomputed features to be passed in as an arg if fast is used. This is primarily for our own reasons / how we're expecting to use this 😝 
- 
I can see the argument for wrangling and combining the two and then have it run just one column but that assumes that we'll just pass in the one feature as `X` if we want the one column method, and that's not what we want! I *did* refactor some underlying logic though to avoid repeats, lmk what ya think!

Reply:
It's all good, I expected there would be a good reason I wasn't immediately privy to!